File: fullstack-demo/server/src/products/products.controller.ts
import { Controller, Get, Post, Body, Param, ParseIntPipe } from '@nestjs/common';
import { ProductsService } from './products.service';
import { CreateProductDto } from './dto/create-product.dto';
import { Product } from './product.entity'; // 🔑 从 Entity 导入

@Controller('products')
export class ProductsController {
  constructor(private readonly productsService: ProductsService) {}

  // GET /products —— 获取列表
  @Get()
  async findAll(): Promise<{ code: number; data: Product[]; message: string }> {
    const data = await this.productsService.findAll(); // 🔑 await 数据库查询
    return { code: 0, data, message: 'success' };
  }

  // GET /products/:id —— 获取单个
  @Get(':id')
  async findOne(
    @Param('id', ParseIntPipe) id: number,
  ): Promise<{ code: number; data: Product; message: string }> {
    const data = await this.productsService.findOne(id);
    return { code: 0, data, message: 'success' };
  }

  // POST /products —— 创建商品
  @Post()
  async create(
    @Body() dto: CreateProductDto,
  ): Promise<{ code: number; data: Product; message: string }> {
    const data = await this.productsService.create(dto);
    return { code: 0, data, message: '创建成功' };
  }
}

File: fullstack-demo/server-java/src/main/java/com/fullstack/demo/products/ProductsController.java
package com.fullstack.demo.products;

import com.fullstack.demo.common.ApiResponse;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 商品控制器 —— 对应 NestJS 的 ProductsController
 * 路由前缀: /products
 */
@RestController
@RequestMapping("/products")
public class ProductsController {

    private final ProductsService productsService;

    public ProductsController(ProductsService productsService) {
        this.productsService = productsService;
    }

    /**
     * GET /products —— 获取商品列表
     */
    @GetMapping
    public ApiResponse<List<Product>> findAll() {
        List<Product> data = productsService.findAll();
        return ApiResponse.success(data);
    }

    /**
     * GET /products/{id} —— 获取单个商品
     */
    @GetMapping("/{id}")
    public ApiResponse<Product> findOne(@PathVariable Long id) {
        Product data = productsService.findOne(id);
        return ApiResponse.success(data);
    }

    /**
     * POST /products —— 创建商品
     */
    @PostMapping
    public ApiResponse<Product> create(@Valid @RequestBody CreateProductDto dto) {
        Product data = productsService.create(dto);
        return ApiResponse.success(data, "创建成功");
    }
}
